include/body-parts/BodyPart.hpp

```diff
@@ -1,120 +1,164 @@
 #ifndef BODY_PART_HPP
 #define BODY_PART_HPP
 
+#include <stack>
 #include <Vector4.hpp>
 #include <vector>
+#include "BodyPartDefines.hpp"
+#include "Matrix4.hpp"
 
 class BodyPart
 {
 public:
-  // Constructor
-  BodyPart() = delete;
-  explicit BodyPart(const Vector4& position, const Vector4& offset);
-  BodyPart(const BodyPart& other) = delete;
+    // Constructors
+    BodyPart();
+
+    // Getters
+    [[nodiscard]] float getDepth() const;
+    [[nodiscard]] float getHeight() const;
+    [[nodiscard]] float getWidth() const;
+    [[nodiscard]] Matrix4 getMatrixStack() const;
+    [[nodiscard]] Matrix4 getTransformationMatrix() const;
+
+    // Setters
+    BodyPart& setColor(short int red, short int green, short int blue);
+    BodyPart& setHeight(float height);
+    BodyPart& setDepth(float depth);
+    BodyPart& setWidth(float width);
+    BodyPart& setParent(BodyPart* parent);
+    BodyPart& setPivotPoint(const Vector4& pivotPoint);
+
+    // Methods
+    BodyPart& rotateX(float angle);
+    BodyPart& rotateY(float angle);
+    BodyPart& rotateZ(float angle);
+    BodyPart& translate(float x, float y, float z);
+    // BodyPart& scale(float x, float y, float z);
+
+    BodyPart& addChild(BodyPart* child);
+    void applyTransformation();
+
+private:
+    /**
+    * The width of the body part.
+    */
+    float _width = LENGTH_BASE_UNIT;
 
-  // Destructor
-  virtual ~BodyPart() = default;
+    /**
+    * The height of the body part.
+    */
+    float _height = LENGTH_BASE_UNIT;
 
-  // Getters
-  [[nodiscard]] Vector4 getOffset() const;
-  [[nodiscard]] Vector4 getPosition() const;
-  [[nodiscard]] double getAngleX() const;
-  [[nodiscard]] double getAngleY() const;
-  [[nodiscard]] double getAngleZ() const;
-  [[nodiscard]] BodyPart* getParent() const;
-  [[nodiscard]] std::vector<BodyPart*> getChildren() const;
+    /**
+    * The depth of the body part.
+    */
+    float _depth = LENGTH_BASE_UNIT;
 
-  // Setters
-  void setOffset(const Vector4& offset);
-  void setPosition(const Vector4& position);
-  void setAngleX(double angleX);
-  void setAngleY(double angleY);
-  void setAngleZ(double angleZ);
-  void setParent(BodyPart* parent);
+    /**
+    * The red value of the body part's color.
+    */
+    short int _red = 255;
 
-  // Operator overloads
-  BodyPart& operator=(const BodyPart& other) = delete;
+    /**
+    * The green value of the body part's color.
+    */
+    short int _green = 255;
 
-  // Methods
-  void addChild(BodyPart* child);
-  void removeChild(BodyPart* child);
-  void updateVertices();
+    /**
+    * The blue value of the body part's color.
+    */
+    short int _blue = 255;
```

Comment:
```suggestion
    unsigned char _blue = 255;
```

on peut mettre unsigned char sur les 3 composantes de la couleur. C'est ce que je voulais faire mais small init c'est entre -32k et 32k environ

Reply:
Pareil